packages/fxa-settings/src/pages/Signin/utils.ts

```diff
@@ -144,24 +143,40 @@ const getNavigationTarget = async ({
   // OAuth redirect can only be obtained when the session is verified
   // otherwise oauth/authorization endpoint throws an "unconfirmed session" error
   if (isOAuth) {
-    const { redirect, code, state } = await finishOAuthFlowHandler(
-      uid,
-      sessionToken,
-      keyFetchToken,
-      unwrapBKey
-    );
-
-    if (integration.isSync()) {
-      firefox.fxaOAuthLogin({
-        action: 'signin',
-        code,
-        redirect,
-        state,
-      });
-      return getSyncNavigate(queryParams);
+    try {
+      const { redirect, code, state } = await finishOAuthFlowHandler(
+        uid,
+        sessionToken,
+        keyFetchToken,
+        unwrapBKey
+      );
+
+      if (integration.isSync()) {
+        firefox.fxaOAuthLogin({
+          action: 'signin',
+          code,
+          redirect,
+          state,
+        });
+        return getSyncNavigate(queryParams);
+      }
+      return { to: redirect, shouldHardNavigate: true };
+    } catch (error) {
+      if (
+        error.errno === AuthUiErrors.TOTP_REQUIRED.errno ||
+        error.errno === AuthUiErrors.INSUFFICIENT_ACR_VALUES.errno
+      ) {
+        return {
+          to: `/inline_totp_setup${queryParams}`,
+          state: createSigninLocationState(),
+        };
+      }
+      // TODO: whatdo otherwise, show error in UI? We should add a try/catch
```

Comment:
A few notes for the reviewer...

The crux of the issue was we were expecting this error to come from the call to signin as every other error we're handling does, but we don't actually know we need to take users to `inline_totp_setup` until we attempt to get the OAuth redirect.

I also noticed these two new inline pages were using `useAccount` in container components, which we want to avoid. (Can't blame anyone for this, we're still doing this in reset password pages.) I was getting errors around reading from Account's apollo cache which were at least coming from the `DataBlock` component anyway, so because of these errors and for consistency and clarity I refactored that component where needed and moved these new inline pages to use "signin state" instead which we're using for our other signin pages.

Because the diff is already large and scope had creeped, I think we're going to want to do what's noted here in a follow up in the prod blocking signin epic. It felt right to handle this `try`/`catch` here, but I've got some Glean metrics pings commented out in a test since those aren't currently hit.  The ticket mentions we want to make sure error states are passed back to reliers - I definitely missed that we needed `returnOnError`, which Barry added for us, and according to a comment in `authorization.js` in Backbone, we use that to report errors back to RPs when `prompt=none` or in this inline flow. I think in this follow up we'll want to look at these things, plus add a `try`/`catch` around other `finishOAuthFlowHandler` calls and handle the errors accordingly.

For now, this at least gets the flow working in React for QA to test.